dashboard/app/controllers/api_controller.rb

```diff
@@ -350,7 +350,7 @@ def load_section
   end
 
   def load_script
-    @script = Script.get_from_cache(params[:script_id]) if params[:script_id].present?
-    @script ||= @section.script || Script.twenty_hour_script
+    script_id = params[:script_id] || @section.script.id
```

Comment:
I don't know, but it seems that `"should get progress for section with section script"` test should cover the case where the `script_id` parameter is not present in any case.